mne/preprocessing/tests/test_stim.py

```diff
@@ -64,13 +66,36 @@ def test_stim_elim():
     test_tminidx = int(0.015 * raw.info['sfreq'])
     test_tmaxidx = int(0.035 * raw.info['sfreq'])
 
-    raw = eliminate_stim_artifact(raw, events, event_id=1, tmin=tmin,
+    raw = fix_stim_artifact_raw(raw, events, event_id=1, tmin=tmin,
                                   tmax=tmax, mode='linear')
     data, times = raw[:, (tidx + test_tminidx):(tidx + test_tmaxidx)]
     diff_data0 = np.diff(data[0])
     diff_data0 -= np.mean(diff_data0)
     assert_array_almost_equal(diff_data0, np.zeros(len(diff_data0)))
-    raw = eliminate_stim_artifact(raw, events, event_id=1, tmin=tmin,
+    raw = fix_stim_artifact_raw(raw, events, event_id=1, tmin=tmin,
                                   tmax=tmax, mode='window')
     data, times = raw[:, (tidx + test_tminidx):(tidx + test_tmaxidx)]
     assert_true(np.all(data) == 0.)
+
+
+def test_stim_fix_evoked():
+    """Test eliminate stim artifact"""
+    raw = Raw(raw_fname, preload=True)
+    events = read_events(event_fname)
+    event_id = 1
+    tmin = -0.02
+    tmax = 0.02
+    picks = pick_types(raw.info, meg=True, eeg=True,
+                       eog=True, stim=False, exclude='bads')
+    epochs = Epochs(raw, events, event_id, tmin, tmax,
+                    picks=picks, preload=False)
+
+    # use window before stimulus
+    evoked = epochs.average()
+    data = evoked.data[:, :]
+    assert_true(np.all(data) == 0.)
```

Comment:
you should test that you get 0 just during tmin/tmax period if using window. Otherwise test that it's linearly interpolated... See how it's done for raw